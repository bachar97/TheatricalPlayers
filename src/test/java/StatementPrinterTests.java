import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.List;

import static org.approvaltests.Approvals.verify;

public class StatementPrinterTests {

    @Test
    void exampleStatement() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new TragedyInfo("Hamlet"));
        plays.put("as-like",  new ComedyInfo("As You Like It"));
        plays.put("othello",  new TragedyInfo("Othello"));
        Invoice invoice = new Invoice(new Customer("1", "BigCo", 103), List.of(
                new Performance("hamlet", 55),
                new Performance("hamlet", 30),
                new Performance("as-like", 20),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        var result = invoice.printText(plays);

        verify(result);
    }
}
