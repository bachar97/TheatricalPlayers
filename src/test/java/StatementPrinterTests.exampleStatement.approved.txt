Invoice for BigCo:
  Hamlet: $650.00 (55 seats)
  Hamlet: $400.00 (30 seats)
  As You Like It: $360.00 (20 seats)
  As You Like It: $580.00 (35 seats)
  Othello: $500.00 (40 seats)
Total Amount: $2,475.00
You earned 51 Fidelity Points.
You earned a discount of $15.00 due to your Fidelity Points!
Your new Fidelity Points Balance is 4.